### ISSUE TITLE
 the loss does not converge

### ISSUE THREAD

**Issue by heqilearning** [balancap/SSD-Tensorflow#216]
Hello, I  fine-tuning on your model, but the loss does not converge, then I first train on a  small dataset then train with a big dataset, but the loss still does not converge any time. Could you please tell me how to deal with the problem? Thank you!

**Comment by XuanheLiu** [comment]
@heqilearning 我的loss值也不收敛，请问您的loss值和map值都是多少？

**Comment by heqilearning** [comment]
你好，我是在voc数据集上基于作者的模型来微调的，但是损失值一直在10-50之间震荡，我改过很多参数来训练但是效果仍然不好，作者是否是在imagenet上训练的呢？另外请问一下你的损失值是多少呢，训练集是什么呢？

**Comment by XuanheLiu** [comment]
@heqilearning My loss value also fluctuates in a large range, and I trained it on voc2007+2012. The author seems to have been trained on Imagenet.

**Comment by gunpowder1473** [comment]
我基于VOC2007从头训练了一个模型，做了一些小的调整，包括可能将原图送入网络以及计算最后loss时将每个GT对应的得分最高的defaultbox算入正样本，最后得到loss在10~20之间


      我没有使用作者的eval，而是比较训练得到到bbox 与 GT box算jaccard系数，大于0.5视为正确，最后计算平均准确率。在这种前提下得到我自己训练的样本准确度为37%，而下载了作者的checkpoint依然使用这种办法，得到的准确度是51%（均在VOC2007测试集）

      使用作者的模型进行fine-tune，发现他的Loss其实是高于我的样本，在20~40之间，训练2w次（VOC2007训练集），总的准确度提高了2%，但对于部分类别的识别率下降了，特别是部分类别有接近10的提升或下降，这正常么？

作者的原始模型：
The mean accuracy is 50.908%, The maximum is 100.000%, The minimum is 0.000%.

The class aeroplane has appered 311 times, and 123 times are wrong, its accuracy is 60.450%

The class bicycle has appered 389 times, and 131 times are wrong, its accuracy is 66.324%

The class bird has appered 576 times, and 290 times are wrong, its accuracy is 49.653%

The class boat has appered 393 times, and 248 times are wrong, its accuracy is 36.896%

The class bottle has appered 657 times, and 512 times are wrong, its accuracy is 22.070%

The class bus has appered 254 times, and 79 times are wrong, its accuracy is 68.898%

The class car has appered 1541 times, and 636 times are wrong, its accuracy is 58.728%

The class cat has appered 370 times, and 89 times are wrong, its accuracy is 75.946%

The class chair has appered 1374 times, and 882 times are wrong, its accuracy is 35.808%

The class cow has appered 329 times, and 181 times are wrong, its accuracy is 44.985%

The class diningtable has appered 299 times, and 146 times are wrong, its accuracy is 51.171%

The class dog has appered 530 times, and 172 times are wrong, its accuracy is 67.547%

The class horse has appered 395 times, and 118 times are wrong, its accuracy is 70.127%

The class motorbike has appered 369 times, and 168 times are wrong, its accuracy is 54.472%

The class person has appered 5227 times, and 2499 times are wrong, its accuracy is 52.191%

The class pottedplant has appered 592 times, and 456 times are wrong, its accuracy is 22.973%

The class sheep has appered 311 times, and 153 times are wrong, its accuracy is 50.804%

The class sofa has appered 396 times, and 183 times are wrong, its accuracy is 53.788%

The class train has appered 302 times, and 104 times are wrong, its accuracy is 65.563%

The class tvmonitor has appered 361 times, and 182 times are wrong, its accuracy is 49.584%



fine-tune作者的模型：
The mean accuracy is 53.666%, The maximum is 100.000%, The minimum is 0.000%.

The class aeroplane has appered 311 times, and 110 times are wrong, its accuracy is 64.630%

The class bicycle has appered 389 times, and 118 times are wrong, its accuracy is 69.666%

The class bird has appered 576 times, and 255 times are wrong, its accuracy is 55.729%

The class boat has appered 393 times, and 232 times are wrong, its accuracy is 40.967%

The class bottle has appered 657 times, and 515 times are wrong, its accuracy is 21.613%

The class bus has appered 254 times, and 72 times are wrong, its accuracy is 71.654%

The class car has appered 1541 times, and 567 times are wrong, its accuracy is 63.206%

The class cat has appered 370 times, and 86 times are wrong, its accuracy is 76.757%

The class chair has appered 1374 times, and 911 times are wrong, its accuracy is 33.697%

The class cow has appered 329 times, and 154 times are wrong, its accuracy is 53.191%

The class diningtable has appered 299 times, and 188 times are wrong, its accuracy is 37.124%

The class dog has appered 530 times, and 124 times are wrong, its accuracy is 76.604%

The class horse has appered 395 times, and 110 times are wrong, its accuracy is 72.152%

The class motorbike has appered 369 times, and 111 times are wrong, its accuracy is 69.919%

The class person has appered 5227 times, and 2450 times are wrong, its accuracy is 53.128%

The class pottedplant has appered 592 times, and 404 times are wrong, its accuracy is 31.757%

The class sheep has appered 311 times, and 148 times are wrong, its accuracy is 52.412%

The class sofa has appered 396 times, and 163 times are wrong, its accuracy is 58.838%

The class train has appered 302 times, and 74 times are wrong, its accuracy is 75.497%

The class tvmonitor has appered 361 times, and 147 times are wrong, its accuracy is 59.280%



我自己从头训练的模型：
The mean accuracy is 37.467%, The maximum is 100.000%, The minimum is 0.000%.

The class chair has appered 1374 times, and 1148 times are wrong, its accuracy is 16.448%

The class car has appered 1541 times, and 702 times are wrong, its accuracy is 54.445%

The class horse has appered 395 times, and 175 times are wrong, its accuracy is 55.696%

The class person has appered 5227 times, and 3007 times are wrong, its accuracy is 42.472%

The class bicycle has appered 389 times, and 188 times are wrong, its accuracy is 51.671%

The class cat has appered 370 times, and 197 times are wrong, its accuracy is 46.757%

The class dog has appered 530 times, and 344 times are wrong, its accuracy is 35.094%

The class train has appered 302 times, and 137 times are wrong, its accuracy is 54.636%

The class aeroplane has appered 311 times, and 173 times are wrong, its accuracy is 44.373%

The class diningtable has appered 299 times, and 210 times are wrong, its accuracy is 29.766%

The class tvmonitor has appered 361 times, and 229 times are wrong, its accuracy is 36.565%

The class bird has appered 576 times, and 443 times are wrong, its accuracy is 23.090%

The class bottle has appered 657 times, and 615 times are wrong, its accuracy is 6.393%

The class motorbike has appered 369 times, and 191 times are wrong, its accuracy is 48.238%

The class pottedplant has appered 592 times, and 513 times are wrong, its accuracy is 13.345%

The class boat has appered 393 times, and 284 times are wrong, its accuracy is 27.735%

The class sofa has appered 396 times, and 249 times are wrong, its accuracy is 37.121%

The class sheep has appered 311 times, and 217 times are wrong, its accuracy is 30.225%

The class cow has appered 329 times, and 216 times are wrong, its accuracy is 34.347%

The class bus has appered 254 times, and 127 times are wrong, its accuracy is 50.000%

**Comment by HiKapok** [comment]
If you still need one state-of-the-art SSD implemetationin Tensorflow, you can check [this one](https://github.com/HiKapok/SSD.TensorFlow) which also got to 77.8%.

**Comment by LJXLJXLJX** [comment]
@heqilearning 你好 我也遇到了相同的问题 用voc2007训练好的模型 在voc2007test下测试 mAP只有0.07
请问你现在解决了吗

**Comment by sumenpuyuan** [comment]
@ gunpowder1473 
can you tell me your wechat？many thanks！

**Comment by sumenpuyuan** [comment]
@gunpowder1473 您好，方便告诉下你的微信号，我看你的git工程中有写你accuracy的代码，我想请教一下，十分感谢！

**Comment by gunpowder1473** [comment]
@sumenpuyuan Gunpowder1605

**Comment by dayL-W** [comment]
@heqilearning 嗨，你好啊，我在VOC_2017数据上训练10w左右后，训练集和测试集的mAP都在0.38左右，不知道怎么调优了，请问你的问题解决了吗？
